### PR TITLE
feat(Tus-1254): Add client remote interfaces

### PR DESCRIPTION
**Pull request by Yangsta911**

### CODE REVIEW

File: pkg/zhonghong_protocol/rtuclient.go [hunk 1]
```diff
@@ -21,14 +21,10 @@ type RTUClientHandler struct {
 
 type ChecksumStruct struct{}
```
Comment (qs-wang) [inline pkg/zhonghong_protocol/rtuclient.go]: no need of a struct

File: pkg/zhonghong_protocol/rtuclient.go [hunk 2]
```diff
@@ -59,23 +55,43 @@ type rtuPackager struct {
 //	Data            : 0 up to 252 bytes
 //	CRC             : 2 byte
```
Comment (qs-wang) [inline pkg/zhonghong_protocol/rtuclient.go]: fix the comment

File: pkg/zhonghong_protocol/rtuclient.go [hunk 3]
```diff
@@ -59,23 +55,43 @@ type rtuPackager struct {
...
 func (mb *rtuPackager) Encode(pdu *ProtocolDataUnit) (adu []byte, err error) {
-	// todo check header to find length
-	length := len(pdu.Data) + 3
-	if length > rtuMaxSize {
-		err = fmt.Errorf("zhonghongprotocol: length of data '%v' must not be bigger than '%v'", length, rtuMaxSize)
+	if pdu.CommandType == "remote" {
```
Comment (qs-wang) [inline pkg/zhonghong_protocol/rtuclient.go]: split the remote control to a separate package

**Review by qs-wang — CHANGES_REQUESTED** [review]

File: pkg/zhonghong_by_wire_protocol/rtuclient_by_wire.go [hunk 4]
```diff
@@ -19,16 +21,11 @@ type RTUClientHandler struct {
...
 }
 
-type ChecksumStruct struct{}
-
-func (c *ChecksumStruct) Sum(a, b int) int {
-	return a + b
-}
 
-func (c *ChecksumStruct) Checksum(data []byte) int {
+func Checksum(data []byte) int {
```
Comment (qs-wang) [inline pkg/zhonghong_by_wire_protocol/rtuclient_by_wire.go]: this is a share function

**Review by qs-wang — CHANGES_REQUESTED** [review]

**Comment by ajaxjiang96** [comment]
I have some suggestions on naming and folder organization

1. Instead of calling them "gateway" and "by_wire" we can just refer them by their models, where "gateway" is B19 and "by_wire" is b27. It's a bit confusing because both of them are connected by wires and are called gateway.
2. Folder names should be the same as the package names
3. Packages should be separated into different folders
4. For those packages you don't want to share with third parties, you may move them into a "internal" folder. By convention other developers won't (not supposed to) quote those packages. Hopefully.
5. No empty files

**Comment by ajaxjiang96** [comment]
Meanwhile package names should be precise and short. You already have the name"zhonghong" in your library name (repo name), you may omit the "zhonghong" in the package names.
The official document encourages to use abbreviations as well. [Link here](https://go.dev/blog/package-names)

**Review by qs-wang — APPROVED** [review]

**Pull request merged**